### ISSUE TITLE
More useful vendor string on Intel GPUs

### ISSUE THREAD

**Issue by breyerml** [AdaptiveCpp/AdaptiveCpp#1090]
**Describe the motivation for the feature request**
Executing `opensycl-info` gives `pci:32902` as vendor string for an integrated Intel Gen11 GPU (which is obviously also the case if you programmatically query the `sycl::info::device::vendor`).

**Describe the solution you'd like**
It should output some more useful vendor string like `NVIDIA` on an NVIDIA GPU.

**If applicable, describe alternatives you've considered**
Currently, we are comparing the vendor string against `pci` to check whether an Intel GPU is available, which is, to be honest, not a permanent solution.

**Additional context**
- `HIPSYCL_VERSION`: [HIPSYCL_VERSION.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142596/HIPSYCL_VERSION.txt)
- `OPENSYCL_INFO`: [OPENSYCL_INFO.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142595/OPENSYCL_INFO.txt)

**Comment by illuhad** [comment]
This might not be easily possible. On CUDA or HIP, we can just assume that all discovered devices are NVIDIA or AMD respectively. On Level Zero, the API is not tied in the same way to a specific vendor (even though the main implementation of Level Zero is for Intel GPUs).
The only way Level Zero has to query the vendor is this here: https://github.com/oneapi-src/level-zero/blob/11c3649a05cf346157bbc6d93a330c33bb7ff7f4/include/ze_api.h#L1383 which only returns the PCI id. We do this, because it's the only thing we can do.

We could special case the Intel PCI id, but it sounds like a hack to me.

**Comment by fodinabor** [comment]
Well, the PCI id mapping -> vendor is pretty well defined:
https://pcisig.com/membership/member-companies?combine=&order=field_vendor_id&sort=asc
So, I'd say it's okay to have a translation function where we can add vendors that we know are supported...?

cf: https://github.com/openbmc/peci-pcie/blob/8b18f526772c96c785a62e3a811eacf24d3b637f/include/pciVendors.hpp#L23
which is generated with: https://github.com/openbmc/peci-pcie/blob/8b18f526772c96c785a62e3a811eacf24d3b637f/utils/GetPCI-SIGVendorIDMap.py

**Comment by breyerml** [comment]
Ahh I didn't know that these values are standardized (if thought it is a more or less random value). 
It would be nice, if hipSYCL could map these values to real names. But if they are standardized we can also do that.

**Comment by illuhad** [comment]
For Intel, it's fairly easy to remember, their PCI id is `0x8086` ;) We do return the number in decimal though.

**Comment by al42and** [comment]
> This might not be easily possible. On CUDA or HIP, we can just assume that all discovered devices are NVIDIA or AMD respectively.

Is `HIP_PLATFORM=nvidia` not supported? :thinking:

**Comment by illuhad** [comment]
@al42and Not really. There are a couple of vestigial lines around this, but I don't expect it to actually work, and I don't see any reason to support this since you can just directly go to CUDA. For example, the HIP and CUDA runtime backends' SSCP JIT currently directly goes to the corresponding llvm-to-backend infrastructure for AMD and NVIDIA respectively.
